manifest-build-tools/application/version_generator.py

```diff
@@ -0,0 +1,200 @@
+#!/usr/bin/env python
+# Copyright 2015-2016, EMC, Inc.
+
+"""
+The script compute the version of a package, just like:
+1.1-1-devel-20160809150908-7396d91
+
+usage:
+./on-tools/manifest-build-tools/HWIMO-BUILD on-tools/manifest-build-tools/application/version_generator.py 
+--repo-dir /home/onrack/rackhd/release/rackhd-repos/PengTian0/b/b/on-http
+--manifest-repo-dir /home/onrack/rackhd/release/rackhd-repos/PengTian0/b/build-manifests
+--is-official-release
+--parameter-file version.txt
+
+Because this script need to import scripts under lib.
+The script HWIMO-BUILD helps to add the scripts under lib to python path.
+
+The required parameters: 
+repo-dir
+manifest-repo-dir
+
+The optional parameters:
+is-official-release (default value is false)
+parameter-file (default value is release_version)
+"""
+import os
+import sys
+import argparse
+import datetime
+import subprocess
+from RepositoryOperator import RepoOperator
+from common import *
+
+class VersionGenerator(object):
+    def __init__(self, repo_dir, manifest_repo_dir):
+        """
+        :return:
+ 
+        """
+        self._repo_dir = repo_dir
+        self._manifest_repo_dir = manifest_repo_dir
+        self.repo_operator = RepoOperator()
+ 
+    def generate_small_version(self):
```

Comment:
can you please add more comment , to tell people what's small_version ,big_version and candidate_version?